providers/autodns/api.go

```diff
@@ -0,0 +1,143 @@
+package autodns
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"sort"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+)
+
+type ZoneListFilter struct {
+	Key string `json:"key"`
+	Value string `json:"value"`
+	Operator string `json:"operator"`
+	Link string `json:"link,omitempty"`
+	Filter []*ZoneListFilter `json:"filters,omitempty"`
+}
+
+type ZoneListRequest struct {
+	Filter []*ZoneListFilter `json:"filters"`
+}
+
+func (api *autoDnsProvider) request(method string, requestPath string, data interface{}) ([]byte, error) {
+	client := &http.Client{}
+
+	requestUrl := api.baseURL
+	requestUrl.Path = api.baseURL.Path + requestPath
+
+	request := &http.Request{
+		URL: &requestUrl,
+		Header: api.defaultHeaders,
+		Method: method,
+	}
+
+	if data != nil {
+		body, _ := json.Marshal(data)
+
+		buffer := bytes.NewBuffer(body)
+
+		request.Body = io.NopCloser(buffer)
+	}
+
+	response, error := client.Do(request)
+
+	if error != nil {
+		return nil, error
+	}
+
+	defer response.Body.Close()
+
+	responseText, _ := ioutil.ReadAll(response.Body)
```

Comment:
Style: remove the blank line after `ReadAll()` and the error handling.

Reply:
Hopefully I fixed this with https://github.com/StackExchange/dnscontrol/pull/1336/commits/50697d7dd6a45eace6e33de2f1678daf828b44bd

Such strict style conventions in Go are something I'm not used to yet.